### PR TITLE
Improve documentation

### PR DESCRIPTION
**Pull request by reachfh**
Improved documentation.
Added dialyxir module and ran dialyzer, correcting invalid specs.

### CODE REVIEW

File: lib/off_broadway/kafka/acknowledger.ex [hunk 1]
```diff
@@ -1,10 +1,11 @@
...
   @moduledoc """
-  Implements the Broadway acknowledger behaviour, handling acking of processed
-  messages back to Kafka once they have been successfully processed. Message
-  ack references are stored in ETS as an ordered set and acknowledgements are
-  performed in the order received rather than the order processed to ensure that
-  a failure of the Broadway pipeline does not allow messages received later but
+  Implements the Broadway acknowledger behaviour, acking
+  messages back to Kafka once they have been successfully processed.
+
+  It stores message ack references in ETS as an ordered set, acknowleging
```
Comment (jdenen) [inline lib/off_broadway/kafka/acknowledger.ex]:
```suggestion
  It stores message ack references in ETS as an ordered set, acknowledging
```

File: lib/off_broadway/kafka/producer.ex [hunk 2]
```diff
@@ -1,23 +1,30 @@
 defmodule OffBroadway.Kafka.Producer do
   @moduledoc """
-  Implements the logic to handle incoming messages through
-  the broadway pipeline. Sends messages to the `handle_info/2`
-  and `handle_demand/2` functions based on requests and tracks
-  acknowledgements in state.
+  Defines a Brodway Producer module which receives messages from Kafka
```
Comment (jdenen) [inline lib/off_broadway/kafka/producer.ex]:
```suggestion
  Defines a Broadway Producer module which receives messages from Kafka
```

Comment (reachfh) [reply lib/off_broadway/kafka/producer.ex]: Ran spellcheck and fixed a bunch of typos. Should have done that before...

**Comment by jdenen** [comment]
Thanks again @reachfh. These changes were just released as part of `1.0.0`.

**Pull request merged**